terraform-module-standards.md

```diff
@@ -0,0 +1,340 @@
+# Terraform Module Development Standards
+This document serves as a guidepost for effective development with Terraform. Our goal is to provide a unified and consistent approach to managing code by outlining some rules of engagement. These following standards cover basic style and structure for your Terraform module configurations.
+
+The key words MUST, MUST NOT, REQUIRED, SHALL, SHALL NOT, SHOULD, SHOULD NOT, RECOMMENDED, MAY, and OPTIONAL in this document are to be interpreted as described in [RFC 2119](https://www.ietf.org/rfc/rfc2119.txt).
+
+## Terraform Modules
+The following guidelines apply to both [Child Modules](#child-modules) and [Root Modules](#root-modules).
+
+Open source modules should be published to the [Terraform Registry](https://registry.terraform.io/).
+
+### Module Structure
+* Terraform modules must follow the [standard module structure](https://developer.hashicorp.com/terraform/language/modules/develop#standard-module-structure).
+* Every module must start with a `main.tf` file, where resources are located by default.
+  * Among other resources, all [locals](https://developer.hashicorp.com/terraform/language/values/locals) should be de ned here in a single locals block.
+* All modules must have a `README.md` (with basic documentation in Markdown format).
+* Examples should be located in `examples/`, each with its own subdirectory and a `README.md` file.
+* Logical groupings of resources can be grouped into their own files and given descriptive names, like `network.tf`, `instances.tf`, or `loadbalancer.tf`.
+  * Avoid giving every resource its own file. Resources should be grouped by shared purpose. For example, “metal_project_api_key” and “metal_project_ssh_key” would likely be combined in `project.tf`.
+* Only Terraform (`*.tf`) and repo metadata files (like `README.md`, `CHANGELOG.md`, or `kitchen.yml`) should exist at the root directory of a module.
+* Additional documentation should be stored in a `docs/` subdirectory.
+
+### Naming Convention
+* All configuration objects should be named using underscores to delimit multiple words. This practice ensures consistency with the naming convention for resource types, data source types, and other prede ned values. Note that this convention does not apply to name [arguments](https://www.terraform.io/docs/configuration/resources.html#resource-types-and-arguments).
```

Comment:
A markdown linter would insert a space between headings and bullet lists. (VS Code has one that will autocorrect these observations)

Reply:
Thanks! This has been addressed on commit 029c82e089218dbf238fc2d3eaf5fe0b2930ce15